src/coreComponents/mesh/CellElementRegion.cpp

```diff
@@ -35,15 +72,157 @@ CellElementRegion::CellElementRegion( string const & name, Group * const parent
+  cellBlocks.forSubGroups< CellBlockABC >( [&] ( CellBlockABC const & cellBlock )
+  {
+    attributeValues.insert( CellElementRegion::getCellBlockAttributeValue( cellBlock.getName() ) );
+  } );
+  return attributeValues;
+}
+
+std::set< string > getCellBlockNamesSet( Group const & cellBlocks )
+{
+  std::set< string > cellBlockNames;
+  cellBlocks.forSubGroups< CellBlockABC >( [&] ( CellBlockABC const & cellBlock )
+  {
+    cellBlockNames.insert( cellBlock.getName() );
+  } );
+  return cellBlockNames;
+}
+
+std::set< string > CellElementRegion::computeSelectedCellBlocks( Group const & cellBlocks ) const
+{
+  std::set< string > selectedCellBlocks;
+
+  GEOS_THROW_IF( m_cellBlockAttributeValues.empty() && m_cellBlockMatchPatterns.empty() && m_cellBlockNames.empty(),
+                 GEOS_FMT( "{}: cellBlocks must be selected to fill this region (using {}, {} or {}).",
+                           getDataContext(),
+                           viewKeyStruct::sourceCellBlockNamesString(),
+                           viewKeyStruct::cellBlockAttributeValuesString(),
+                           viewKeyStruct::cellBlockMatchPatternsString() ),
+                 InputError );
+  GEOS_THROW_IF( ( !m_cellBlockAttributeValues.empty() + !m_cellBlockMatchPatterns.empty() + !m_cellBlockNames.empty() ) != 1,
+                 GEOS_FMT( "{}: Only one setting must be used to select cellBlocks ({}, {} or {}).",
+                           getDataContext(),
+                           viewKeyStruct::sourceCellBlockNamesString(),
+                           viewKeyStruct::cellBlockAttributeValuesString(),
+                           viewKeyStruct::cellBlockMatchPatternsString() ),
+                 InputError );
+
+  {
+    std::set< string > cellBlockMatchPatterns;
 
-  for( string const & cellBlockName : this->m_cellBlockNames )
+    // for each desired attribute values, we add the following the match patterns:
+    // Attribute value, followed by an underscore, followed by one or more characters.
+    static string_view constexpr attributeValuePatternSuffix = "_?*";
+    for( integer attributeValue : m_cellBlockAttributeValues )
+    {
+      cellBlockMatchPatterns.insert( std::to_string( attributeValue ) + string( attributeValuePatternSuffix ) );
+    }
+
+    for( string const & matchPattern : m_cellBlockMatchPatterns )
+    {
+      cellBlockMatchPatterns.insert( matchPattern );
+    }
+
+    for( string const & matchPattern : cellBlockMatchPatterns )
+    {
+      bool matching = false;
+      cellBlocks.forSubGroups< CellBlockABC >( [&] ( CellBlockABC const & cellBlock )
+      {
+        // if the pattern matches the tested cellBlock name
+        if( fnmatch( matchPattern.c_str(), cellBlock.getName().c_str(), 0 ) == 0 )
+        {
+          matching = true;
+          selectedCellBlocks.insert( cellBlock.getName() );
+        }
+      } );
+
+      // a) check for the case where the user is requesting attribute values
```

Comment:
Disputable, but what about a
```c++
if( matching ) continue;
```
?

Reply:
I usually try to avoid them but here it seems bound to the errors.
And what do you think of a:
```c++
if( !matching )
{
  // error checking
}
```
It adds the brackets lines but prevent anyone to fail to write code under this error checking part. I hope I'm being clear.

Reply:
That does not really matter. It's just that the `matching` boolean was filled but it's usage was a bit hidden.
As you wish.